git/git.go

```diff
@@ -10,20 +10,29 @@ import (
 	"regexp"
 	"strings"
 
-	"github.com/cli/cli/utils"
+	"github.com/cli/cli/internal/run"
 )
 
 func VerifyRef(ref string) bool {
 	showRef := exec.Command("git", "show-ref", "--verify", "--quiet", ref)
-	err := utils.PrepareCmd(showRef).Run()
+	err := run.PrepareCmd(showRef).Run()
 	return err == nil
 }
 
 // CurrentBranch reads the checked-out branch for the git repository
 func CurrentBranch() (string, error) {
+	err := run.PrepareCmd(GitCommand("log")).Run()
+	if err != nil {
+		// this is a hack.
+		errRe := regexp.MustCompile("your current branch '([^']+)' does not have any commits yet")
```

Comment:
All very fair criticisms; I knew it was a bad solution but figured you'd have a good suggestion on a different approach ^_^() I just don't have enough lower level git knowledge yet.

`git symbolic-ref` exists as far back as git 1.9 and I like using it more than the other options. I was going to suggest we switch to that by default instead of the existing `rev-parse` invocation but it errors on detached HEAD :(

I'll clean up `CurrentBranch` to use both, I guess.